dotnet/WPF/VisualLayerIntegration/VisualLayerIntegration/BarGraphUtility/BarBrushHelper.cs

```diff
@@ -30,111 +30,92 @@
+                linearGradientBrush.ColorStops.Add(stop);
             }
 
-            var brushes = new CompositionBrush[numBrushes];
-            for (int j = 0; j < numBrushes; j++)
-            {
-                brushes[j] = lgb;
-            }
-            return brushes;
+            return linearGradientBrush;
         }
 
-        internal CompositionBrush[] GenerateAmbientAnimatingPerBarLinearGradient(int numBrushes, List<Color> colors)
+        internal CompositionBrush CreateAnimatedLinearGradientBrushes(Color[] colors)
         {
-            var lgb = compositor.CreateLinearGradientBrush();
-            lgb.RotationAngleInDegrees = 45;
+            var linearGradientBrush = _compositor.CreateLinearGradientBrush();
+            linearGradientBrush.RotationAngleInDegrees = 45;
 
-            int i = 0;
-            var animationDuration = 100;
+            // Set long animation duration so that animated effect is slow
+            var animationDuration = TimeSpan.FromSeconds(100);
+
+            var i = 0;
             foreach (Color color in colors)
             {
-                float offset = i / ((float)colors.Count - 1);
+                var offset = i / ((float)colors.Length - 1);
 
-                var stop = compositor.CreateColorGradientStop(offset, color);
-                lgb.ColorStops.Add(stop);
-                InitLGBAnimation(stop, animationDuration, 1.0f);
+                var stop = _compositor.CreateColorGradientStop(offset, color);
+                linearGradientBrush.ColorStops.Add(stop);
+                AnimateLinearGradientstopOffset(stop, animationDuration, 1);
 
                 // Create a second mirrored stop for all colors but the first.
                 if (offset > 0)
                 {
-                    var stop2 = compositor.CreateColorGradientStop(-offset, color);
-                    lgb.ColorStops.Add(stop2);
-                    InitLGBAnimation(stop2, animationDuration, 1.0f);
+                    var stop2 = _compositor.CreateColorGradientStop(-offset, color);
+                    linearGradientBrush.ColorStops.Add(stop2);
+                    AnimateLinearGradientstopOffset(stop2, animationDuration, 1);
                 }
 
                 i++;
             }
 
-            var brushes = new CompositionBrush[numBrushes];
-            for (int j = 0; j < numBrushes; j++)
-            {
-                brushes[j] = lgb;
-            }
-
-            return brushes;
+            return linearGradientBrush;
         }
 
-        private void InitLGBAnimation(CompositionColorGradientStop stop, int duration, float offsetAdjustment)
+        private void AnimateLinearGradientstopOffset(CompositionColorGradientStop stop, TimeSpan duration, float offsetAdjustment)
         {
-            var animateStop = compositor.CreateScalarKeyFrameAnimation();
+            var animateStop = _compositor.CreateScalarKeyFrameAnimation();
             animateStop.InsertKeyFrame(0.0f, stop.Offset);
```

Comment:
For consistency... any time you pass an integer (e.g. 0, 1, 2,... any whole number) to somewhere expecting a float or double, there's no need for the .0f.